compiler/AST/checkAST.cpp

```diff
@@ -247,6 +247,7 @@ void checkPrimitives()
      case PRIM_CAPTURE_FN:
      case PRIM_CREATE_FN_TYPE:
      case PRIM_STRING_COPY:
+     case PRIM_MEMCPY:
```

Comment:
It doesn't seem like we should need a primitive for `memcpy`, would
```chpl
extern proc memcpy(destination: c_ptr, const source: c_ptr, num: size_t);
```

not work? `c_ptr(void)` will codegen as `typedef void *c_ptr_void`. I'd rather go that direction than adding more special support in the compiler for this stuff. We could `typedef c_void_ptr = c_ptr(void)` as well in chapel as well.